Source/WTF/wtf/posix/FileSystemPOSIX.cpp

```diff
@@ -270,15 +271,17 @@ std::pair<String, PlatformFileHandle> openTemporaryFile(StringView prefix, Strin
     // This is OK for now since the code using it is built on macOS only.
     ASSERT_UNUSED(suffix, suffix.isEmpty());
 
-    char buffer[PATH_MAX];
-    if (snprintf(buffer, PATH_MAX, "%s/%sXXXXXX", temporaryFileDirectory(), prefix.utf8().data()) >= PATH_MAX)
-        goto end;
+    const char* tmp = temporaryFileDirectory();
+    CString pre = prefix.utf8();
+    size_t len = strlen(tmp) + 1 + pre.length() + 6 + 1;
+    auto buffer = MallocSpan<char>::malloc(len);
+    snprintf(buffer.mutableSpan().data(), len, "%s/%sXXXXXX", tmp, pre.data());
```

Comment:
I noticed that this will malfunction if the prefix happens to end with an "X", which we could fix by including at least one non-X character before "XXXXXX".